execution-frameworks/Invoke-AtomicRedTeam/Invoke-AtomicRedTeam.ps1

```diff
@@ -33,75 +30,127 @@ Github repo: https://github.com/redcanaryco/atomic-red-team
 #>
 
 function Confirm-Dependencies {
+    [CmdletBinding(DefaultParameterSetName = 'dependencies',
+        SupportsShouldProcess = $true,
+        ConfirmImpact = 'Medium')]
+    Param ( )
+    
+    Write-Verbose -Message 'Checking whether powershell-yaml is installed'
+
+    try {
+        if (-not(Get-Module -Name 'powershell-yaml' -ListAvailable)) {
+            if ($pscmdlet.ShouldProcess("PowerShell-Yaml Module", "Install required module")) {
+                Install-Module -Name powershell-yaml -Force
+                Write-Verbose -Message 'Successfully installed powershell-yaml module'
+            }
+        }
+    }
+    catch {
+        Write-Error -ErrorRecord $Error[0] -RecommendedAction 'Please install powershell-yaml before continuing'
+        throw 'Unable to install powershell-yaml'
+    }
 
-    if (Get-Module -ListAvailable -Name 'powershell-yaml') {
-        Write-Host "PowerShell-Yaml Module Exists. Good To Go." -Foreground Green
+    try {
+        Write-Verbose -Message 'Importing powershell-yaml module'
+        Import-Module -Name powershell-yaml -Force
     }
-    else {
-        $Title = "PowerShell-Yaml Is Not Installed"
-        $Message = "Do you want to Install?"
-        $Yes = New-Object System.Management.Automation.Host.ChoiceDescription "&Y-Yes", "Yes"
-        $No = New-Object System.Management.Automation.Host.ChoiceDescription "&N-No", "No"
-        $Options = [System.Management.Automation.Host.ChoiceDescription[]]($Yes, $No)
-        $Result = $Host.ui.PromptForChoice($Title, $Message, $Options, 0)
-
-        switch ($Result) {
-            0 { Install-Module -Name powershell-yaml }
-            1 { Write-Host "Atomic Red Team Requires PowerShell-Yaml Exiting"; exit}
-        }
+    catch {
+        Write-Error -ErrorRecord $Error[0] -RecommendedAction 'Please import the powershell-yaml module before continuing'
+        throw 'Unable to import the powershell-yaml module'
     }
 }
 
 function Get-AtomicTechnique {
-    [CmdletBinding()]
+    [CmdletBinding(DefaultParameterSetName = 'technique',
+        SupportsShouldProcess = $true,
+        PositionalBinding = $false,
+        ConfirmImpact = 'Medium')]
     Param(
+        [Parameter(Mandatory = $true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = $true,
+            ParameterSetName = 'technique')]
+        [ValidateNotNull()]
```

Comment:
This is extraneous unless I'm remembering how `ValidateNotNullOrEmpty` work.

Reply:
You are correct!  My mistake!  I'll remove that.

Reply:
This is resolved.